### ISSUE TITLE
Vulnerability in https-proxy-agent < 3.0.0

### ISSUE THREAD

**Issue by anajavi** [web-push-libs/web-push#519]
# Setup

*Operating System*: OS X
*Node Version*: 12.13
*web-push Version*: 3.4.0

- [ ] Chrome
- [ ] Firefox
- [ ] Opera for Android
- [ ] Samsung Internet Browser
- [ X ] Other

# Problem

npm audit complains about vulnerability in dependency of web-push:
```
                       === npm audit security report ===                        
                                                                                
┌──────────────────────────────────────────────────────────────────────────────┐
│                                Manual Review                                 │
│            Some vulnerabilities require your attention to resolve            │
│                                                                              │
│         Visit https://go.npm.me/audit-guide for additional guidance          │
└──────────────────────────────────────────────────────────────────────────────┘
┌───────────────┬──────────────────────────────────────────────────────────────┐
│ High          │ Machine-In-The-Middle                                        │
├───────────────┼──────────────────────────────────────────────────────────────┤
│ Package       │ https-proxy-agent                                            │
├───────────────┼──────────────────────────────────────────────────────────────┤
│ Patched in    │ >=3.0.0                                                      │
├───────────────┼──────────────────────────────────────────────────────────────┤
│ Dependency of │ web-push                                                     │
├───────────────┼──────────────────────────────────────────────────────────────┤
│ Path          │ web-push > https-proxy-agent                                 │
├───────────────┼──────────────────────────────────────────────────────────────┤
│ More info     │ https://npmjs.com/advisories/1184                            │
└───────────────┴──────────────────────────────────────────────────────────────┘
```

# Expected

No security warnings

# Features Used

# Example / Reproduce Case

```shell
npm install --save web-push
npm audit
```

# Other

This is already fixed in master, so releasing master should fix this.

**Comment by ecc521** [comment]
I'm not quite sure how web-push uses ```https-proxy-agent```, but this may well be a very relevant, and quite serious issue. (Interception of messages, which would be used for 2-factor-auth or something).

**Comment by marco-c** [comment]
Should be fixed by fb2c1c24b8ad8a9e1d042783a4fb255f868b4862, I'll release a new version.

**Comment by marco-c** [comment]
> I'm not quite sure how web-push uses https-proxy-agent

It is used when the `proxy` option is used.
